frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -650,6 +661,147 @@ describe("AddToDashSelectDashModal", () => {
         });
       });
     });
+
+    describe("search dashboards", () => {
+      describe("questions in the root collection (public collection)", () => {
+        it("should search questions only in public collections", async () => {
+          await setup({
+            card: CARD_IN_ROOT_COLLECTION,
+          });
+
+          const typedText = "question";
+          const dashboardInPublicCollection = createMockSearchResult({
+            name: "dashboard in public collection",
+            model: "dashboard",
+          });
+          fetchMock.get(
+            {
+              url: "path:/api/search",
+              query: {
+                models: "dashboard",
+                q: typedText,
+              },
+            },
+            {
+              data: [dashboardInPublicCollection],
+            },
+          );
+
+          userEvent.click(screen.getByRole("button", { name: "Search" }));
+          userEvent.type(
+            screen.getByPlaceholderText("Search"),
+            "question{enter}",
+          );
+
+          expect(
+            await screen.findByText(dashboardInPublicCollection.name),
+          ).toBeInTheDocument();
+
+          // There's no way to math a URL that a query param is not present
+          // with fetch-mock, so we have to assert it manually.
+          const call = fetchMock.lastCall("path:/api/search");
+          const urlObject = new URL(checkNotNull(call?.request?.url));
+          expect(urlObject.pathname).toEqual("/api/search");
+          expect(
+            urlObject.searchParams.get("filter_items_in_personal_collection"),
+          ).toEqual(null);
+        });
+      });
+
+      describe("questions in public collections", () => {
```

Comment:
Tests under this category are basically the same as tests under **questions in the root collection (public collection)**. But I added **questions in the root collection (public collection)** because previously there are some special cases for the root collection, so I want to have them the same, even though they're not different for searching dashboards.

Reply:
That could work!

Reply:
I tried doing that, but seems like there are exceptions for the root collection.

In the root collection we can see both public and personal collections e.g.
- public1
- public2
- my personal collection
- all personal collections (if you're an admin)

but in other public collection e.g. inside the `public1` there's no way to have personal collection here at all, so there are 1 less test to test (tests that no personal collections are rendered). And a lot of expectations would be different e.g. breadcrumbs and to assert the collections that are rendered under each starting collection. So there would be a lot of parameters that I think would make the test less readable than the version we're having right now.

This is similar to the concern I have put [in this comment](https://github.com/metabase/metabase/pull/35215/files/9b1afe4f0746e4ca54554641b1d083490090801e#r1384977958).